File: Chapter 3/collatz.py
def collatz(number):
    if number % 2 == 0:
        return number // 2
    else:
        return 3 * number + 1
    
while True:
    print('Input number')
    n = input()
    try:
        n = int(n)
    except:
        print('Please enter an integer')
        continue
    while n != 1:
        n = collatz(n)
        print(n)
    break

File: Chapter 3/Collatz.java
class Collatz {
    Collatz(int number) {
        while (number != 1) {
            number = collatz(number);
            System.out.println(number); 
        }
    }
    int collatz(int num) {
        if (num % 2 == 0) {
            return num / 2;
        } else {
            return 3 * num + 1;
        }
    }
}
